### PR TITLE
fix: update bot naming rules

### LINKED ISSUE #3185: Bot's with dots in the name is accepted but breaks LUIS

### PR DESCRIPTION
**Pull request by liweitian**
## Description
do not allow using dot in bot's dialog and trigger name.
## Task Item
Closes #3185 
## Screenshots

### CODE REVIEW

**Comment by beyackle** [comment]
LGTM, but I agree this could use some simple unit tests.

**Comment by liweitian** [comment]
test cases added

File: Composer/packages/client/__tests__/components/CreationFlow/DefineConversation/index.test.tsx [hunk 1]
```diff
@@ -82,4 +82,14 @@ describe('<DefineConversation/>', () => {
         'https://raw.githubusercontent.com/microsoft/botframework-sdk/master/schemas/component/component.schema',
     });
   });
+
+  it('disable submit button', async () => {
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/__tests__/components/CreationFlow/DefineConversation/index.test.tsx:86]: Try to make these strings more descriptive of the behavior it is asserting.

> 'does not allow submission when the name is invalid'

File: Composer/packages/client/__tests__/components/createDialogModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+describe('<CreateDialogModal/>', () => {
+  let onSubmitMock;
+  let onDismissMock;
+  let component, storeContext;
+  function renderComponent() {
+    return render(
+      <StoreContext.Provider value={storeContext}>
+        <CreateDialogModal isOpen onDismiss={onDismissMock} onSubmit={onSubmitMock} />
+      </StoreContext.Provider>
+    );
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/__tests__/components/createDialogModal.test.tsx:19]: Use the `renderWithStore` utility function.

File: Composer/packages/client/__tests__/components/createDialogModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    onSubmitMock = jest.fn();
+  });
+
+  it('should render the component', () => {
+    component = renderComponent();
+    expect(component.container).toBeDefined();
+  });
+
+  it('disable submit button', async () => {
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/__tests__/components/createDialogModal.test.tsx:38]: See my comment above.

File: Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx [hunk 4]
```diff
@@ -37,7 +37,7 @@ import { StoreContext } from '../../store';
 
 import { styles, dropdownStyles, dialogWindow, intent } from './styles';
 
-const nameRegex = /^[a-zA-Z0-9-_.]+$/;
+const nameRegex = /^[a-zA-Z0-9-_]+$/;
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/components/ProjectTree/TriggerCreationModal.tsx:40]: This should probably be hoisted into a shared module that can be consumed by both places where it is used.

File: Composer/packages/client/__tests__/components/createDialogModal.test.tsx (hunk 2 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/__tests__/components/createDialogModal.test.tsx:19]: Still need the `renderWithStore` utility here.

File: Composer/packages/client/__tests__/components/createDialogModal.test.tsx (hunk 3 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/client/__tests__/components/createDialogModal.test.tsx:38]: This hasn't been updated yet.

**Comment by cwhitten** [comment]
@liweitian  please address conflicts

**Pull request merged**